### PR TITLE
[Machine-learning][ACR-1722][ACR-1723] Update ml-service documentation

### PR DESCRIPTION
**Pull request by anyj0527**
- Update the guide documentation of Machine Learning Service API
- ACR: ACR-1722, ACR-1723

Signed-off-by: Yongjoo Ahn <yongjoo1.ahn@samsung.com>

### CODE REVIEW

File: docs/application/native/guides/machine-learning/overview.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ You can use the following machine learning features in your native applications:
 
 - [Machine Learning Service](machine-learning-service.md)
 
-  The Machine Learning Service API provides interfaces to store and fetch the pipeline description for AI application developers.
+  You can register and deploy specific AI operations as ML service and let other applications use this service via the network.
```
Comment (safir-srbd) [inline docs/application/native/guides/machine-learning/overview.md:18]:
```suggestion
  You can register and deploy specific AI operations as an ML service and let other applications use this service via the network.
```

File: docs/application/native/guides/machine-learning/machine-learning-service.md [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
-- Delete the pipeline description.
+The main features of the Machine Learning Service API include the following:
+
+- Providing AI pipelines as ML services via machine-learning-agent (daemon).
+
+  AI developers can manage their AI pipeline description with a unique name. Then they can launch the pipeline as an ML service.
+
+- Using launched ML services via the network.
+
+  Application developers can request their data to be processed by the launched ML service. The communication is powered by [NNStreamer's tensor_query](https://nnstreamer.github.io/gst/nnstreamer/tensor_query/README.html) which implements [Edge-AI](https://nnstreamer.github.io/edge-ai.html) functionality.
```
Comment (safir-srbd) [inline docs/application/native/guides/machine-learning/machine-learning-service.md:13]:
```suggestion
  Application developers can request their data to be processed by the launched ML service. The communication is powered by [NNStreamer's tensor_query](https://nnstreamer.github.io/gst/nnstreamer/tensor_query/README.html), which implements [Edge-AI](https://nnstreamer.github.io/edge-ai.html) functionality.
```

File: docs/application/native/guides/machine-learning/machine-learning-service.md [hunk 3]
```diff
@@ -23,47 +29,167 @@ Follow the steps below to enable your application to use Machine Learning Servic
...
+tensor_transform mode=arithmetic option=typecast:float32,add:0.0,div:255.0 !
+queue ! tensor_filter framework=tensorflow2-lite model=/path/to/yolov5s/model/file.tflite !
+other/tensors,num_tensors=1,types=float32,format=static,dimensions=85:6300:1:1,framerate=0/1 !
+tensor_decoder mode=bounding_boxes option1=yolov5 option2=/path/to/coco/label.txt option4=320:320 option5=320:320 !
+video/x-raw,format=RGBA,width=320,height=320 ! tensor_converter !
+other/tensors,format=static,num_tensors=1,dimensions=4:320:320:1,types=uint8 !
+tensor_query_serversink async=false sync=false
+```
+
+The pipeline expects a 320x320 RGB raw image as input, and pre-process it to invoke the yolov5s tflite model. The inference result is decoded to bounding box image with tensor_decoder. The service provider can upgrade their internal neural network without changing the code of client if input and output layout is preserved.
```
Comment (safir-srbd) [inline docs/application/native/guides/machine-learning/machine-learning-service.md:48]:
```suggestion
The pipeline expects a 320x320 RGB raw image as input, and pre-process it to invoke the yolov5s tflite model. The inference result is decoded to bounding box image with tensor_decoder. The service provider can upgrade their internal neural network without changing the code of the client if input and output layout is preserved.
```

File: docs/application/native/guides/machine-learning/machine-learning-service.md [hunk 4]
```diff
@@ -23,47 +29,167 @@ Follow the steps below to enable your application to use Machine Learning Servic
...
 
-The following code snippet shows a simple example of the Machine Learning Service API.
+Clients who want to use this service may use their own pipeline communicating with the service, or can use Machine Learning Service API which is more easy to use.
```
Comment (safir-srbd) [inline docs/application/native/guides/machine-learning/machine-learning-service.md:50]:
```suggestion
Clients who want to use this service may use their own pipeline communicating with the service, or can use the Machine Learning Service API, which is more easy to use.
```

File: docs/application/native/guides/machine-learning/machine-learning-service.md [hunk 5]
```diff
@@ -23,47 +29,167 @@ Follow the steps below to enable your application to use Machine Learning Servic
    <feature name="http://tizen.org/feature/machine_learning.service">true</feature>
    ```
 
-## An example of usage
+## An example of object detection with Machine Learning Service API
+
+Let some AI developers want to provide a object detection service. The ML service takes an image data from other applications, and give back the image with bounding boxes drawn on it. An example pipeline can be described as this:
```
Comment (safir-srbd) [inline docs/application/native/guides/machine-learning/machine-learning-service.md:34]:
```suggestion
Let's say some AI developers want to provide an object detection service. The ML service takes an image data from other applications, and give back the image with bounding boxes drawn on it. An example pipeline can be described as this:
```

File: docs/application/native/guides/machine-learning/machine-learning-service.md [hunk 6]
```diff
@@ -23,47 +29,167 @@ Follow the steps below to enable your application to use Machine Learning Servic
...
 
-The following code snippet shows a simple example of the Machine Learning Service API.
+Clients who want to use this service may use their own pipeline communicating with the service, or can use Machine Learning Service API which is more easy to use.
 
-An ML developer adds a pipeline description which includes preprocessing and invoking a neural network model, and an application developer can get the registered pipeline.
+The following code snippet shows an example of how the Machine Learning Service API can be used in this scenario. Note that checking return value and error handling is omitted for brevity.
+
+### AI developers launching the object detection ML service
+
+AI developers can manage the pipeline description.
```
Comment (safir-srbd) [inline docs/application/native/guides/machine-learning/machine-learning-service.md:56]:
```suggestion
AI developers can manage the pipeline description by doing the following:
```
Introduce a list with a complete sentence, not a partial one that's completed by the list items.
Reference: https://developers.google.com/style/lists#introductory-sentences-for-lists

File: docs/application/native/guides/machine-learning/machine-learning-service.md [hunk 7]
```diff
@@ -23,47 +29,167 @@ Follow the steps below to enable your application to use Machine Learning Servic
...
-    error:
-    ml_pipeline_destroy (handle);
-    g_free (pipeline);
+    ml_service_get_pipeline ("object_detection_service", &pipeline);
+
+    // Delete the pipeline description of given service.
+    ml_service_delete_pipeline ("object_detection_service");
+    ```
+
+AI developers can manage the ML service.
```
Comment (safir-srbd) [inline docs/application/native/guides/machine-learning/machine-learning-service.md:87]:
```suggestion
AI developers can manage the ML service by doing the following:
```
Introduce a list with a complete sentence, not a partial one that's completed by the list items.
Reference: https://developers.google.com/style/lists#introductory-sentences-for-lists

File: docs/application/native/guides/machine-learning/machine-learning-service.md [hunk 8]
```diff
@@ -23,47 +29,167 @@ Follow the steps below to enable your application to use Machine Learning Servic
...
+    // Stop the given service.
+    ml_service_stop_pipeline (object_detection_service_h);
+
+    // Destroy the given service.
+    ml_service_destroy (object_detection_service_h);
+    ```
+
+### Application developers using the service
+
+Clients (application developers) can use services with the Machine Learning Service API exploiting NNStreamer's tensor_query feature.
```
Comment (safir-srbd) [inline docs/application/native/guides/machine-learning/machine-learning-service.md:116]:
```suggestion
Clients (application developers) can use services with the Machine Learning Service API exploiting NNStreamer's tensor_query feature. They can do this by using the following steps:
```
An ordered/unordered list should be preceded by a “colon”.

**Review by safir-srbd — COMMENTED** [review]
@anyj0527 Hello. Hope you are well.

I have reviewed the files in this PR and provided some suggestions. Please incorporate them when possible.

**Comment by anyj0527** [comment]
@safir-srbd Thanks for your careful review. I've reflected all your suggestions, PTAL.

**Review by safir-srbd — APPROVED** [review]
@anyj0527 Thank you for making the implemented changes.

@mijin85cho @asm-foysal I am approving this PR for merging.

**Pull request merged**